src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/VectorScoreScript.java

```diff
@@ -0,0 +1,149 @@
+package com.amazon.opendistroforelasticsearch.knn.plugin.script;
+
+import com.amazon.opendistroforelasticsearch.knn.index.util.KNNConstants;
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.util.BytesRef;
+import org.elasticsearch.script.ScoreScript;
+import org.elasticsearch.search.lookup.SearchLookup;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * Vector score script used for adjusting the score based on similarity space
+ * on a per document basis.
+ */
+public class VectorScoreScript extends ScoreScript {
+
+    private BinaryDocValues binaryDocValuesReader;
+    private final float[] inputVector;
+    private final String similaritySpace;
+
+    public float l2Squared(float[] queryVector, float[] inputVector) {
+        long squaredDistance = 0;
+        for (int i = 0; i < inputVector.length; i++) {
+            squaredDistance += Math.pow(queryVector[i]-inputVector[i], 2);
+        }
+        return squaredDistance;
+    }
+
+    /**
+     * This function called for each doc in the segment. We evaluate the score of the vector in the doc
+     *
+     * @param explanationHolder A helper to take in an explanation from a script and turn
+     *                          it into an {@link org.apache.lucene.search.Explanation}
+     * @return score of the vector to the query vector
+     */
+    @Override
+    public double execute(ScoreScript.ExplanationHolder explanationHolder) {
+        float score = Float.MIN_VALUE;
+        try {
+            float[] doc_vector;
+            BytesRef bytesref = binaryDocValuesReader.binaryValue();
+            // If there is no vector for the corresponding doc then it should be not considered for nearest
```

Comment:
In what case does this happen?

Reply:
If for some reason, some documents do not have vector field then we need to handle this case gracefully.
I am doing some more tests around this.